### ISSUE TITLE
[BUG] Rancher is restarted in loop after a restore

### ISSUE THREAD

**Issue by ldevulder** [rancher/rancher#44613]
**Rancher Server Setup**
- Rancher version: 2.9-head
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): v1.26.10+k3s2
   - Command used to install Rancher:
   ```bash
   $ helm upgrade --install rancher rancher-latest/rancher \
       --namespace cattle-system \
       --create-namespace \
       --set hostname=<MY_HOSTNAME> \
       --set bootstrapPassword=rancherpassword \
       --set extraEnv[0].name=CATTLE_SERVER_URL \
       --set extraEnv[0].value=https://<MY_HOSTNAME> \
       --set rancherImageTag=v2.9-head \
       --set replicas=1 \
       --devel
   ```
   **NOTE:** some options like `CATTLE_SERVE_URL` are set because the deployment code is used in `Elemental` CI.
- Proxy/Cert Details: selfsigned CA (default Rancher installation)

**Information about the Cluster**
- Kubernetes version:  v1.26.10+k3s2
- Cluster Type (Local/Downstream): Local
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? Admin

**Describe the bug**

A simple K3s cluster for Local is deployed, then `rancher-backup-operator` v5.0.0-rc2 is installed. I triggered a backup (called `test-backup`) on local storage (default configuration of K3s, nothing more) and then I triggered a restore from the previously created backup resource.

All seems to be restore but not the Rancher pod, it runs in loop. I attached the restore logs. I did a quick check and I can see error like this one:
```
INFO[2024/02/29 13:23:52] restoreResource: Restoring ui-preferred of type management.cattle.io/v3, Resource=settings 
INFO[2024/02/29 13:23:52] Successfully restored ui-preferred
INFO[2024/02/29 13:23:52] restoreResource: Restoring monitoring-ui-view of type management.cattle.io/v3, Resource=roletemplates
ERRO[2024/02/29 13:23:52] Error restoring resource monitoring-ui-view of type management.cattle.io/v3, Resource=roletemplates: restoreResource: err updating resource Internal error occurred: failed calling webhook "rancher.cattle.io.roletemplates.management.cattle.io": failed to call webhook: an error on the server ("{\"kind\":\"AdmissionReview\",\"apiVersion\":\"admission.k8s.io/v1\",\"request\":{\"uid\":\"06e02a3e-4a31-4b23-881a-2bd680f7e020\",\"kind\":{\"group\":\"management.cattle.io\",\"version\":\"v3\",\"kind\":\"RoleTemplate\"},\"resource\":{\"group\":\"management.cattle.io\",\"version\":\"v3\",\"resource\":\"roletemplates\"},\"requestKind\":{\"group\":\"management.cattle.io\",\"version\":\"v3\",\"kind\":\"RoleTemplate\"},\"requestResource\":{\"group\":\"management.cattle.io\",\"version\":\"v3\",\"resource\":\"roletemplates\"},\"name\":\"monitoring-ui-view\",\"operation\":\"UPDATE\",\"userInfo\":{\"username\":\"system:serviceaccount:cattle-resources-system:rancher-backup\",\"uid\":\"fbafcf3f-097a-4927-9724-a1e3d42c0fa2\",\"groups\":[\"system:serviceaccounts\",\"system:serviceaccounts:cattle-resources-system\",\"system:authenticated\"],\"extra\":{\"authentication.kubernetes.io/pod-name\":[\"rancher-backup-58746db55f-x6dqs\"],\"authentication.kubernetes.io/pod-uid\":[\"383afb45-9f20-4f0a-bcae-f0b48eb4cb77\"]}},\"object\":{\"apiVersion\":\"management.cattle.io/v3\",\"builtin\":true,\"context\":\"project\",\"description\":\"\",\"displayName\":\"View Monitoring\",\"external\":true,\"hidden\":false,\"kind\":\"RoleTemplate\",\"metadata\":{\"annotations\":{\"cleanup.cattle.io/rtUpgradeCluster\":\"true\",\"lifecycle.cattle.io/create.mgmt-auth-roletemplate-lifecycle\":\"true\"},\"creationTimestamp\":\"2024-02-29T13:16:56Z\",\"finalizers\":[\"controller.cattle.io/mgmt-auth-roletemplate-lifecycle\"],\"generation\":1,\"labels\":{\"authz.management.cattle.io/bootstrapping\":\"default-roletemplate\"},\"managedFields\":[{\"apiVersion\":\"management.cattle.io/v3\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:builtin\":{},\"f:context\":{},\"f:description\":{},\"f:displayName\":{},\"f:external\":{},\"f:hidden\":{},\"f:metadata\":{\"f:annotations\":{\".\":{},\"f:cleanup.cattle.io/rtUpgradeCluster\":{},\"f:lifecycle.cattle.io/create.mgmt-auth-roletemplate-lifecycle\":{}},\"f:finalizers\":{\".\":{},\"v:\\\"controller.cattle.io/mgmt-auth-roletemplate-lifecycle\\\"\":{}},\"f:labels\":{\".\":{},\"f:authz.management.cattle.io/bootstrapping\":{}}}},\"manager\":\"rancher\",\"operation\":\"Update\",\"time\":\"2024-02-29T13:17:03Z\"}],\"name\":\"monitoring-ui-view\",\"res") has prevented the request from succeeding
INFO[2024/02/29 13:23:52] restoreResource: Restoring azure-group-cache-size of type management.cattle.io/v3, Resource=settings 
INFO[2024/02/29 13:23:52] Successfully restored azure-group-cache-size
INFO[2024/02/29 13:23:52] restoreResource: Restoring tls-ciphers of type management.cattle.io/v3, Resource=settings 
INFO[2024/02/29 13:23:52] Successfully restored tls-ciphers
```

**NOTE:** even if the issue has been found in Elemental CI, I redo manual testing from scratch and a simple configuration without any Elemental components fails, so this is definitely not related to Elemental.

**To Reproduce**

Simply install Rancher Manager 2.9-head on K3s (v1.26.10) and install the `rancher-backup-operator` v5.0.0-rc2. Then do a backup and try to restore directly right after.

**Result**

**Expected Result**

A fully restored and working Rancher Manager.

**Screenshots**

**Additional context**

[Restore logs](https://github.com/rancher/rancher/files/14448714/rancher-2.9_backup-restore-operator.zip)

**Comment by kkaempf** [comment]
@snasovich - is that something Hostbuster can look into ?

**Comment by snasovich** [comment]
@kkaempf , thank you for surfacing it. At a glance it looks like something to be looked into by Observation&Backup team initially or maybe even Frameworks team as per log above `monitoring-ui-view` roletemplate fails to restore due to Webhook issue.
FYI @prachidamle @MKlimuszka  - assigning both teams to it for now.

However, this may not be the root cause of the issue. @ldevulder / @kkaempf , could you please provide more information on "Rancher pod, it runs in loop" - i.e. logs from Rancher pod from its start to its termination?

**Comment by ldevulder** [comment]
> @ldevulder / @kkaempf , could you please provide more information on "Rancher pod, it runs in loop" - i.e. logs from Rancher pod from its start to its termination?

I'll have to redo a full test on my lab for this, will try next week. But I already provided some logs with error in it, did you check them? Also easy to reproduce if you have an env ;-) But as I said I will try to redo it again next week.

**Comment by ldevulder** [comment]
@snasovich here 3 logs I was able to get from the restarting rancher pod. It seems to be killed, don't know if you can find something useful in it.
[rancher-1.log](https://github.com/rancher/rancher/files/14742715/rancher-1.log)
[rancher-2.log](https://github.com/rancher/rancher/files/14742716/rancher-2.log)
[rancher-3.log](https://github.com/rancher/rancher/files/14742717/rancher-3.log)

**Comment by ldevulder** [comment]
FYI I tested with latest released version (v5.0.0-rc6) of `rancher-backup-restore` plugin for Rancher Manager 2.9  and it now works as expected. So it has been fixed between v5.0.0-rc2 and v5.0.0-rc6, but unfortunately version rc3, rc4 and rc5 cannot be installed, so I'm not able to test each version to try to find what exactly fixed the issue.

Anyway, as it works now I close this issue.
